op-proposer/metrics/metrics.go

```diff
@@ -98,3 +107,7 @@ const (
 func (m *Metrics) RecordL2BlocksProposed(l2ref eth.L2BlockRef) {
 	m.RecordL2Ref(BlockProposed, l2ref)
 }
+
+func (m *Metrics) RecordL1GasFee(receipt *types.Receipt) {
```

Comment:
Gas fee tracking shouldn't be part of the `RefMetrics`, that's purely for reference data.

We just did a refactor in the `op-batcher` and `op-proposer` to move more transaction handling into the `op-service/txmgr`.  So this metric (together with a few other metrics) should probably be added to the `txmgr` so that it's available to both, the batcher and proposer. It is on our radar to add metrics there. We can sync on that :)

Reply:
oh nice. just saw the refactored code on `txmgr`. I am happy to help with this. Let me know if you have some guidance for me about what ways you are expecting. I can wait till both `op-batcher` and `op-proposer` go via the new `txmgr` codebase and then add metrics. Let me know :)

Reply:
That sounds great, I'll write up an issue for the txmgr metrics and let you know.

Reply:
Thanks @sebastianst

Reply:
Hey @shrimalmadhur I've got one more set of large change to the internals of the txmgr here (https://github.com/ethereum-optimism/optimism/pull/5286, still WIP), but is shouldn't change the constructor/initialization flow.

Reply:
Thanks @sebastianst ! I will start working on it. I will close this PR and start a different one just to make things cleaner. 

@trianglesphere 👍